src/api/controllers/notificationController.js

```diff
@@ -0,0 +1,170 @@
+import status from '../../helpers/constants/status.codes';
+import sendError from '../../helpers/error.sender';
+import errorMessages from '../../helpers/constants/error.messages';
+import emailUnsubscriptionConfig from '../../helpers/constants/email.unsubscription.config.json';
+
+const { NotificationConfig, Notification } = models;
+
+/**
+ * A class to handle all notifications
+ */
+export default class NotificationController {
+  /**
+   * A method to get user configs
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} return an object containing set configuration
+   */
+  static async getConfig(req, res) {
+    // Initializing variables
+    const result = {};
+    const {
+      user: { id }
+    } = req.user;
+
+    try {
+      const userConfig = await NotificationConfig.findOne({
+        userId: id
+      });
+
+      userConfig.config = Object.keys(userConfig).length ? JSON.parse(userConfig.config) : null;
+
+      // Sending the result
+      result.status = status.OK;
+      result.config = userConfig;
+      return res.status(status.OK).json(result);
+    } catch (error) {
+      sendError(status.NOT_FOUND, res, 'config', errorMessages.configNotFound);
+    }
+  }
+
+  /**
+   * A method to update user configs
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} return an object containing set configuration
+   */
+  static async updateConfig(req, res) {
+    // Initializing variables
+    const result = {};
+    const {
+      user: { id }
+    } = req.user;
+
+    const isUnsubscribeRequest = req.url.includes('email');
+
+    // Updating the configuration
+    if (isUnsubscribeRequest) {
+      await NotificationConfig.update({
+        config: JSON.stringify(emailUnsubscriptionConfig)
+      },
+      {
+        where: {
+          userId: id
+        }
+      });
+    } else {
+      await NotificationConfig.update({
+        config: JSON.stringify(req.body)
+      },
+      {
+        where: {
+          userId: id
+        }
+      });
+    }
+
+    // sending the result back
+    result.status = status.OK;
+    result.message = 'Configurations updated successfully';
+
+    return res.status(status.CREATED).json({ result });
+  }
+
+  /**
+   * A method to get all notifications of a user
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} return an object containing all notifications
+   */
+  static async getAll(req, res) {
+    // Initializing variables
+    const result = {};
+    const {
+      user: { id }
+    } = req.user;
+
+    const isUnseenRequest = req.url.includes('unseen');
+    let notifications;
+
+    if (isUnseenRequest) {
+      notifications = await Notification.findAll({
+        where: {
+          userId: id,
+          status: 'unseen'
+        }
+      });
+    } else {
+      notifications = await Notification.findAll({
+        where: {
+          userId: id,
+          status: 'seen'
+        }
+      });
+    }
```

Comment:
according to the description this method is supposed to send all user's notifications but the logic is about seen or unseen notifications

Reply:
Copy-paste issue bro.
I'm gonna change the description